### PR TITLE
Adds Publish Event Activity Tests

### PR DESCRIPTION
**Pull request by sfmskywalker**
This change is [Reviewable](https://reviewable.io/reviews/elsa-workflows/elsa-core/7093)

### CODE REVIEW

File: test/unit/Elsa.Activities.UnitTests/Event/PublishEventTests.cs [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        var publisher = Substitute.For<IEventPublisher>();
+
+        // Act
+        await ExecuteAsync(activity, publisher);
+
+        // Assert
+        await publisher.Received(1).PublishAsync(
+            eventName,
+            null,
+            Arg.Any<string?>(),
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply test/unit/Elsa.Activities.UnitTests/Event/PublishEventTests.cs]: Great catch! You're absolutely right, this is a legitimate issue.

The current implementation only normalizes empty strings (`""`), not whitespace. This is inconsistent behavior that could lead to bugs when users accidentally pass whitespace-only correlation IDs.

I'll implement your suggested fix:

1. Add a NullIfWhiteSpace() extension method to StringExtensions
2. Update PublishEvent.ExecuteAsync to use it instead of NullIfEmpty()
3. Add test coverage for the whitespace case

This ensures whitespace correlation IDs are properly normalized to null, making the behavior more intuitive and consistent.

File: test/component/Elsa.Workflows.ComponentTests/Scenarios/Activities/Primitives/Event/PublishEventTests.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    [Fact]
+    public async Task PublishEvent_LocalEvent_CompletesWorkflow()
+    {
+        // Act
+        var result = await _workflowRunner.RunAndAwaitWorkflowCompletionAsync(WorkflowDefinitionHandle.ByDefinitionId(PublishAndConsumeEventWorkflow.DefinitionId, VersionOptions.Published));
+
+        // Assert
+        Assert.Equal(WorkflowSubStatus.Finished, result.WorkflowExecutionContext.SubStatus);
```
Comment (j03y-nxxbz) [inline test/component/Elsa.Workflows.ComponentTests/Scenarios/Activities/Primitives/Event/PublishEventTests.cs:39]: Should there also be an Assert.Equal(WorkflowStatus.Finished, result.WorkflowExecutionContext.Status)?

File: test/component/Elsa.Workflows.ComponentTests/Scenarios/Activities/Primitives/Event/PublishEventTests.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    [Fact]
+    public async Task PublishEvent_WithPayload_CompletesSuccessfully()
+    {
+        // Act
+        var result = await _workflowRunner.RunAndAwaitWorkflowCompletionAsync(
+            WorkflowDefinitionHandle.ByDefinitionId(PublishGlobalEventWorkflow.DefinitionId, VersionOptions.Published));
+
+        // Assert
+        Assert.Equal(WorkflowSubStatus.Finished, result.WorkflowExecutionContext.SubStatus);
```
Comment (j03y-nxxbz) [inline test/component/Elsa.Workflows.ComponentTests/Scenarios/Activities/Primitives/Event/PublishEventTests.cs]: Same here as my previous comment?

Comment (j03y-nxxbz) [reply test/component/Elsa.Workflows.ComponentTests/Scenarios/Activities/Primitives/Event/PublishEventTests.cs]: In addition, should we do any assert on the Payload? I could only deduce a Payload exists when looking at the workflow definition itself, but I am not seeing what we expect to happen with the payload.

File: test/component/Elsa.Workflows.ComponentTests/Scenarios/Activities/Primitives/Event/PublishEventTests.cs (hunk 2 above)
Comment (sfmskywalker) [reply test/component/Elsa.Workflows.ComponentTests/Scenarios/Activities/Primitives/Event/PublishEventTests.cs:39]: Not necessarily - the main status is derived from the sub status. That logic itself, however, would be good to have covered via some other unit tests (it may or may not be there already).

File: test/component/Elsa.Workflows.ComponentTests/Scenarios/Activities/Primitives/Event/PublishEventTests.cs (hunk 3 above)
Comment (sfmskywalker) [reply test/component/Elsa.Workflows.ComponentTests/Scenarios/Activities/Primitives/Event/PublishEventTests.cs]: That's a great idea, yes we should definitely have a test case that asserts a sent payload is received.

Comment (sfmskywalker) [reply test/component/Elsa.Workflows.ComponentTests/Scenarios/Activities/Primitives/Event/PublishEventTests.cs]: Done 👍🏻
